### ISSUE TITLE
LoggerProxy breaking contract with Psr\Log\LoggerInterface

### ISSUE THREAD

**Issue by Irinina** [magento/magento2#38583]
### Preconditions and environment

- Magento version 2.4.7-beta3

The class `\Magento\Framework\Logger\LoggerProxy` is not compatible with the interface, it is implementing (`Psr\Log\LoggerInterface`). In `2.4.7-beta3`, types were added to the arguments of the implemented functions. This breaks the contract and we get a fatal error. I'm not 100% sure, when the error comes up. For me, it occurs when running `bin/magento setup:upgrade`. But it could be, that something needs to be logged during this, to occur.

### Steps to reproduce

1. run `bin/magento setup:upgrade` 

### Expected result

Command runs without error.

### Actual result

Command breaks with error:
```
Fatal error: Declaration of Magento\Framework\Logger\LoggerProxy::emergency(Stringable|string $message, array $context = []): void must be compatible with Psr\Log\LoggerInterface::emergency($message, array $context = []) in /var/www/html/vendor/magento/framework/Logger/LoggerProxy.php on line 110
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by Bashev** [comment]
Fixed in https://github.com/magento/magento2/pull/38524

@Irinina add `"psr/log": "^2 || ^3"` to your root composer.json as temporary solution. Probably some of your 3th-party modules will be not compatible with `psr/log` `> 1`.

Yireo/webp2 is already compatible.
Smile/elasticsuite will be released soon but you can fix it following instructions in this comment: https://github.com/Smile-SA/elasticsuite/pull/3215#issuecomment-2036927512

**Comment by hostep** [comment]
@engcom-Delta: this got fixed by: https://github.com/magento/magento2/pull/38524, despite it not being marked as merged, the changes actually got included in 2.4.7

So I believe this issue can be closed, unless @Irinina disagrees?

**Comment by Irinina** [comment]
@hostep Yes, I agree that this is fixed in 2.4.7, thanks 👍🏼
